packages/precommit-hooks/default-hooks.yaml

```diff
@@ -0,0 +1,48 @@
+# Pre-commit hook configuration file
+#
+# This contains the list of hooks to be run before committing code.
+# The devsecops-hooks baseline hook is temporarily disabled while we investigate issues with it but will eventually become compulsory.
+# Other hooks maybe added or removed as needed to suit individual project requirements.
+
+repos:
+  #. Temporary disabled while we investigate issues with it
+  #  - repo: https://github.com/ministryofjustice/devsecops-hooks
+  #    rev: v1.1.0
+  #    hooks:
+  #      - id: baseline
+  - repo: local
+    hooks:
+      - id: gitleaks
+        name: Scan commit for secrets
+        language: system
+        entry: gitleaks git --pre-commit --redact --staged --verbose --config .gitleaks/config.toml --gitleaks-ignore-path .gitleaks/.gitleaksignore
+        require_serial: true
+        pass_filenames: false
+      - id: lint
+        name: linting code
+        language: system
+        entry: npm run lint
+        types: [ts]
+        require_serial: true
+        pass_filenames: false
+      - id: typecheck
+        name: verify types
+        language: system
+        entry: npm run typecheck
+        types: [ts]
+        require_serial: true
+        pass_filenames: false
+      - id: test
+        name: running tests
+        language: system
+        entry: npm run test
+        types: [ts]
+        require_serial: true
+        pass_filenames: false
+  - repo: builtin
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-json
+      - id: check-yaml
+      - id: check-merge-conflict
```

Comment:
might be overkill but I threw in a load of the out of the box checks (including the end of file fixer 🥳 ).